nixos/modules/services/networking/seafile.nix

```diff
@@ -0,0 +1,272 @@
+            ${pkgs.sqlite}/bin/sqlite3 ${dataDir}/seafile.db ".read ${pkgs.seafile-server}/share/seafile/sql/seafile.sql"
+        fi
+      '';
+    };
+
+    systemd.services.seahub =
+      let penv = (pkgs.python3.withPackages (ps: with ps; [ gunicorn seahub ]));
+      in {
+        description = "Seafile Server Web Frontend";
+        wantedBy = [ "seafile.target" ];
+        partOf = [ "seafile.target" ];
+        after = [ "network.target" "seaf-server.service" ];
+        requires = [ "seaf-server.service" ];
+        restartIfChanged = true;
+        restartTriggers = [ (seafileConf) ];
+        environment = {
+          PYTHONPATH =
+            "${pkgs.python3Packages.seahub}/thirdpart:${pkgs.python3Packages.seahub}:${penv}/${python.sitePackages}";
+          DJANGO_SETTINGS_MODULE = "seahub.settings";
+          CCNET_CONF_DIR = ccnetDir;
+          SEAFILE_CONF_DIR = dataDir;
+          SEAFILE_CENTRAL_CONF_DIR = centralConfigDir;
+          SEAFILE_RPC_PIPE_PATH = "/run/seafile";
+        };
+        serviceConfig = {
+          User = "seafile";
+          Group = "seafile";
+          DynamicUser = true;
+          ProtectHome = true;
+          PrivateUsers = true;
+          PrivateDevices = true;
+          ProtectClock = true;
+          ProtectHostname = true;
+          ProtectProc = "invisible";
+          ProtectKernelModules = true;
+          ProtectKernelTunables = true;
+          ProtectKernelLogs = true;
+          ProtectControlGroups = true;
+          RestrictNamespaces = true;
+          RestrictAddressFamilies = [ "AF_UNIX" ];
+          LockPersonality = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+          MemoryDenyWriteExecute = true;
+          SystemCallArchitectures = "native";
+          RuntimeDirectory = "seahub";
+          StateDirectory = "seafile";
+          LogsDirectory = "seafile";
+          ExecStart = ''
+            ${penv}/bin/gunicorn seahub.wsgi:application \
+            --name seahub \
+            --workers ${toString cfg.workers} \
+            --log-level=info \
+            --preload \
+            --timeout=1200 \
+            --limit-request-line=8190 \
+            --bind unix:/run/seahub/gunicorn.sock
+          '';
+        };
+        preStart = ''
+          if [ ! -d "${seahubDir}" ]; then
+              mkdir ${seahubDir}
+              ln -sf ${seahubSettings} ${centralConfigDir}/seahub_settings.py
+              # link media into local seahubDir, for avatars
+              mkdir -p ${seahubDir}/media
+              for m in ${pkgs.python3Packages.seahub}/media/*/ ; do
+                ln -s $m ${seahubDir}/media/
+              done
+              rm ${seahubDir}/media/avatars
+              cp -rf ${pkgs.python3Packages.seahub}/media/avatars ${seahubDir}/media/
+              chmod -R 0755 ${seahubDir}/media/avatars
```

Comment:
What is going on here?

Reply:
I reworked the avatar setup to avoid this kind of messy permission fix